integration-libs/epd-visualization/schematics/add-epd-visualization/index.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  chain,
+  noop,
+  Rule,
+  SchematicContext,
+  Tree,
+} from '@angular-devkit/schematics';
+import {
+  addLibraryFeature,
+  addPackageJsonDependenciesForLibrary,
+  CLI_EPD_VISUALIZATION_FEATURE,
+  CustomConfig,
+  EPD_VISUALIZATION_CONFIG,
+  readPackageJson,
+  shouldAddFeature,
+  SPARTACUS_EPD_VISUALIZATION,
+  validateSpartacusInstallation,
+} from '@spartacus/schematics';
+import { parse, stringify } from 'comment-json';
+import { peerDependencies } from '../../package.json';
+import {
+  EPD_VISUALIZATION_FEATURE_NAME_CONSTANT,
+  EPD_VISUALIZATION_FOLDER_NAME,
+  EPD_VISUALIZATION_MODULE,
+  EPD_VISUALIZATION_MODULE_NAME,
+  EPD_VISUALIZATION_ROOT_MODULE,
+  EPD_VISUALIZATION_TRANSLATIONS,
+  EPD_VISUALIZATION_TRANSLATION_CHUNKS_CONFIG,
+  SCSS_FILE_NAME,
+  SPARTACUS_EPD_VISUALIZATION_ROOT,
+  SPARTACUS_EPD_VISUALIZATION_ASSETS,
+} from '../constants';
+import { Schema as SpartacusEpdVisualizationOptions } from './schema';
+
+const TSConfigFileName = 'tsconfig.json';
+
+export function addEpdVisualizationFeature(
+  options: SpartacusEpdVisualizationOptions
+): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const packageJson = readPackageJson(tree);
+    validateSpartacusInstallation(packageJson);
+
+    return chain([
+      addPackageJsonDependenciesForLibrary(peerDependencies, options),
+
+      shouldAddFeature(CLI_EPD_VISUALIZATION_FEATURE, options.features)
+        ? chain([addEpdVisualization(options), updateTsConfig()])
+        : noop(),
+    ]);
+  };
+}
+
+function addEpdVisualization(options: SpartacusEpdVisualizationOptions): Rule {
+  const customConfig: CustomConfig[] = [
+    {
+      import: [
+        {
+          moduleSpecifier: SPARTACUS_EPD_VISUALIZATION_ROOT,
+          namedImports: [EPD_VISUALIZATION_CONFIG],
+        },
+      ],
+      content: `<${EPD_VISUALIZATION_CONFIG}>{
+        epdVisualization: {
+          ui5: {
+            bootstrapUrl: "https://sapui5.hana.ondemand.com/1.97.0/resources/sap-ui-core.js"
+          },
+
+          usageIds: {
```

Comment:
We usually provide default configs in our OOTB modules, hiding the complexity from customers, using the helper `provideDefaultConfig({...})`, e.g. in the feature's root module. I think we should move there the config for `usageIds` and `visualPicking`, instead of pasting it into customer's app's code via schematics. Of course, later on customers can overwrite our default config by providing custom config via `provideConfig({...})` in their app, if only they need.

Reply:
There is some configuration that could reasonably go in default configuration.

The `folderUsageId` configuration is something that we would like to be more dynamic. The original idea behind this was for this to be set on a per deployment environment basis (so that it could have a value of `Online` for the production environment and a value of `Staging` for the staging environment for instance).

However I assume that this is not currently easy to achieve based on the current status of this issue:
https://github.com/SAP/spartacus/issues/5772

Currently a customer would have to created different Spartacus builds for their different environments (manually changing this value for each) in order to achieve this.

Reply:
>Currently a customer would have to created different Spartacus builds for their different environments (manually changing this value for each) in order to achieve this.
Correct.

That said, do you plan to move any more property to the default config? If yes, which?

Reply:
I have moved the configuration for  `usageIds` and `visualPicking` into default configuration.